### DISCUSSION TITLE
Rainforest is visually indistinguishable from regular grass

### DISCUSSION THREAD

**Discussion by supermop** [OpenTTD/OpenTTD#11733] [Features]
### Version of OpenTTD

1.12.0

### Expected result

A visual indication of which tiles are rainforest regions, and thus can have rainforest trees or the tropic sawmill industry.

### Actual result

Rainforest currently uses the same tile sprites as regular grass and rough land, so it is not visually identifiable.  Ideally the tiles would look different, and perhaps less grassy. I would suggest re-using the 2nd stage grass regrowth sprites if no new sprites are wanted.

### Steps to reproduce

Query grass and rainforest tiles in the game.

**Comment by ldpl** [comment]
But why does it need to be distinguishable? Lumber mill placement is the only influence it has in the game and that's part of a larger problem of unclear industry placement rules in general and so should be solved generally, not just for a single industry type. Besides, you can already find rainforest tiles with tile inspection tool, by checking trees, or on the vegetation minimap.

Also, at least one argument against this would be that newgrfs don't have a way to get the ground sprite and have to calculate it themselves. So changing the logic behind it would complicate such calculations and require updating all the affected newgrfs.

P.S. Fun fact, all generated water tiles in tropic are technically rainforest, so it's not just ground.

**Comment by supermop** [comment]
> But why does it need to be distinguishable? Lumber mill placement is the only influence it has in the game and that's part of a larger problem of unclear industry placement rules in general and so should be solved generally, not just for a single industry type. Besides, you can already find rainforest tiles with tile inspection tool, by checking trees, or on the vegetation minimap.
> 
> Also, at least one argument against this would be that newgrfs don't have a way to get the ground sprite and have to calculate it themselves. So changing the logic behind it would complicate such calculations and require updating all the affected newgrfs.
> 
> P.S. Fun fact, all generated water tiles in tropic are technically rainforest, so it's not just ground.

One of the the issues I have is that using the query tool or trying to identify the trees is sort of clunky and counterintuitive vs a visual difference. Depending on the tree sprites in your base set or tree grf, they might not look that different than the regular trees. In addition to the lumber mill placement, its also unclear what trees you can plant where in game.

It's not really clear how rainforest is determined - in the scenario editor it seems that bare land regrows as grass, but if you plant one of the rainforest trees on it, it becomes rainforest? Does rainforest organically grow in this way too? In game, on the other hand, you can't build a rainforest tree on grass - it just silently fails to build a tree, so you have to then check the tile and guess that the tree you wanted was actually rainforest only.  I would instead expect that I can paint/convert land to and from rainforest like with desert in the scenario editor, and then the correct trees will grow?

For potential broader use, other NewGRF industries or houses could choose to allow or forbid placing on rainforest, so I do think there can be broader use in making it easy to visually tell apart, but it's not gamebreakingly critical...

**Comment by 2TallTyler** [comment]
> in the scenario editor it seems that bare land regrows as grass, but if you plant one of the rainforest trees on it, it becomes rainforest?

Yes, and then that tile stays rainforest even if you bulldoze the tree. You have to make it desert to reset the tile. Rainforest does not grow organically — planting rainforest trees is the only way to create it manually.

I would support making rainforest tiles more visually distinct, and possibly even adding a button in Scenario Editor to paint them, as we have with desert tiles.

**Comment by 2TallTyler** [comment]
@zephyris What do you think about this? If we gave you a different grass tile type to draw in rainforest, could you make it look distinct?

**Comment by zephyris** [comment]
Yeah, should be fairly simple to make a specific ground tile. Making something to match OG wouldnb easy enough too.

It'd take a bit of thought about what would be the best design... More brown would work, like fallen leaves. Fallen branches/logs/tree stumps could work, but might look repetitive.

**Comment by EmperorJake** [comment]
OpenGFX2's temperate grass in tropical has grown on me, perhaps that or something similar could be used for rainforest, to indicate the green lushness of the land?

The green grass clashes somewhat when it is next to the desert sand, so it would be good to be able to keep the existing tropical grass as a transition between the lush rainforest and the arid desert.

![image](https://github.com/OpenTTD/OpenTTD/assets/68182631/7b3ba935-351b-41cc-9def-dc661b267174)

**Comment by nielsmh** [comment]
I would support a feature for basesets to be able to supply a ground sprite set for rainforest and rainforest-transition tiles. The transition tiles would probably be used under similar circumstances to the desert-transition tiles. Possibly this requires both plains-to-rainforest and desert-to-rainforest transition sprites?
